hw03_frequency_analysis/top.go

```diff
@@ -1,6 +1,61 @@
 package hw03frequencyanalysis
 
-func Top10(_ string) []string {
-	// Place your code here.
-	return nil
+import (
+	"sort"
+	"strings"
+)
+
+func Top10(inputString string) []string {
+	topStrings := make([]string, 0)
+
+	if inputString == "" {
+		return topStrings
+	}
+
+	strInputSlice := strings.Fields(inputString)
+	wordCounts := make(map[string]int)
+
+	for _, word := range strInputSlice {
+		_, ok := wordCounts[word]
+		if !ok {
+			wordCounts[word] = 1
+		} else {
+			wordCounts[word]++
+		}
```

Comment:
Можно не проверять наличие слова в мапе и сразу делать инкремент

Reply:
@agneum 

```suggestion
		_, ok := wordCounts[word]
		if ok {
			wordCounts[word]++
		}
```

в этом случае падают тесты с ошибкой 

```suggestion
		--- FAIL: TestTop10 (0.00s)
    --- PASS: TestTop10/no_words_in_empty_string (0.00s)
    --- FAIL: TestTop10/positive_test (0.00s)
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0
```

что-то не пойму в чем дело

Reply:
короче переписал я метод вроде норм